### PR TITLE
Docs update for JobPodsReady in GA

### PR DESCRIPTION
**Pull request by mimowo**
Tracking issue: https://github.com/kubernetes/enhancements/issues/2879

### CODE REVIEW

**Comment by katcosgrove** [comment]
Hi, @mimowo! v1.29 Docs Lead here. Please take a look at [Documenting for a release - PR Ready for Review](https://kubernetes.io/docs/contribute/new-content/new-features/#pr-ready-for-review) to get your PR ready for review before Tuesday 14th November 2023. Thank you!

**Comment by mimowo** [comment]
/assign @alculquicondor @sftim 
Minimal changes for now, I could not see any existing section on the feature. Maybe we should add one?

**Comment by alculquicondor** [comment]
No need. We decided that the API reference should be enough, for such a minimal feature.

**Review by alculquicondor — COMMENTED** [review]
/lgtm

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md [hunk 1]
```diff
@@ -267,6 +265,9 @@ For a reference to old feature gates that are removed, please refer to
 | `IPTablesOwnershipCleanup` | `true` | GA | 1.28 | |
 | `InTreePluginRBDUnregister` | `false` | Alpha | 1.23 | 1.27 |
 | `InTreePluginRBDUnregister` | `false` | Deprecated | 1.28 | |
+| `JobReadyPods` | `false` | Alpha | 1.23 | 1.23 |
+| `JobReadyPods` | `true` | Beta | 1.24 | |
```
Comment (tengqm) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md:269]:
```suggestion
| `JobReadyPods` | `true` | Beta | 1.24 | 1.28 |
```

**Pull request merged**